server/vehicle_token_handlers.go

```diff
@@ -0,0 +1,44 @@
+package server
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/dexidp/dex/api/v2"
+)
+
+type vehicleIDTokenClaims struct {
+	Issuer   string   `json:"iss"`
+	Subject  string   `json:"sub"`
+	Audience audience `json:"aud,omitempty"`
+	Expiry   int64    `json:"exp"`
+	IssuedAt int64    `json:"iat"`
+	UserID   string   `json:"userId"`
+
+	Privileges []int64 `json:"privileges,omitempty"`
+}
+
+func (d dexAPI) GetVehiclePrivilegeToken(ctx context.Context, req *api.GetVehiclePrivilegeTokenReq) (*api.GetVehiclePrivilegeTokenResp, error) {
+	expiry := d.serverConfig.Now().Add(time.Minute * 10) // TODO - Discuss an appropriate time
+	v := vehicleIDTokenClaims{
+		Issuer:     d.serverConfig.Issuer,
+		Subject:    req.VehicleTokenId,
```

Comment:
I think `sub` usually indicates the user. Maybe `sub` is the user id and we have some novel claim like `vehicleTokenId`?

@zer0stars do you have any opinions on the shape of this thing?

Reply:
I believe the vehicle is the subject here, since the claims we are generating is about the vehicle.
<img width="739" alt="Screenshot 2022-11-23 at 2 18 19 PM" src="https://user-images.githubusercontent.com/111298305/203631007-3b64d357-288f-4736-b172-22427618172d.png">

According to the docs [https://www.rfc-editor.org/rfc/rfc7519#section-4](https://www.rfc-editor.org/rfc/rfc7519#section-4)

Reply:
I think no matter what we decide, Yev will have some other idea and that's what we'll do. I think just use whatever you want, it's just labels for now.

Reply:
Ok got it

Reply:
I looked at the [OpenID](https://openid.net/specs/openid-connect-core-1_0.html#IDToken) before commenting. To be honest I think these specs are bound to be a bad match for this.

I think it's fine to merge and move on.